tsfc/parameters.py

```diff
@@ -21,6 +21,7 @@
 
     # Precision of float printing (number of digits)
     "precision": numpy.finfo(NUMPY_TYPE).precision,
+    "optimisation": "O2"
```

Comment:
I don't particularly like the idea of calling optimisation levels/modes `O2`, `O3`, etc. These are the traditional command line options for C compilers (and some other compilers), but:

- TSFC has no command line interface, only a Python interface.
- Considering various finite element form compilers/representations/optimisers, I think it is more appropriate to use the term 'optimisation mode' instead of 'optimisation level'. Level suggest that there is an ordering, and that higher levels are clearly more optimised than lower levels. But this is not the case here. "Is tensor representation more optimised than vanilla quadrature? Sometimes..."